Fix non-ground nil handling for assoc operators having different element/list types (#205)
cvc5/ethos: NEWS.md

  treated as a valid code point.
- Fixes issue in the parser which did not guard for overflow of 32 bit unsigned
  values.

> - Fixes an issue where non-ground nil terminators would not be properly computed for operators having types where the element type was different from the return type of the operator.

Reviewer: ```suggestion
- Fixes an issue where non-ground nil terminators would not be properly computed for list construction operators with types where the element type is different from the return type of the operator.
```
